### PR TITLE
Add example for The Things Network

### PR DESCRIPTION
**Pull request by mbgg**
This adds a small program to show how to connect the device to The Things Network backend.

### CODE REVIEW

**Comment by Citrullin** [comment]
@mbgg What do you expect as payload?
Tried: 
test
0x1
1h

But the program is just responding with: 
```
error: argument --payload/-p: invalid hex value:
```

**Comment by mbgg** [comment]
you are right. I'll fix that and send a new pull request.

**Comment by Citrullin** [comment]
You can also just fix it in your branch. It will automatically updated in this PR.:)

**Comment by mbgg** [comment]
Changes in this version:
- you can now use a custom payload with -p 0xaffeaffe
- appuei short parameter changed to -u
- added print of payload to send

**Comment by Citrullin** [comment]
@mbgg Wouldn't make it sense to use ascii which is then converted to hex?

**Comment by mbgg** [comment]
@Citrullin we convert ascii to int in the argparse type lambda. Problem is, that I didn't find how to print hex values using %.16x, that's why I store it as int. BTW casts to int in line 46-50 shouldn't be needed anymore.

**Comment by Citrullin** [comment]
@mbgg I am going to take a look into it on Wednesday or Thursday.

**Comment by WA4OSH** [comment]
In the example. I would send a string of hex representing three channels all Cayenne encoded: 1. Temperature at 20 deg c 2. Humidity 50% and 3. GPS location for Beaverton. OR.  Stretch goal:  a.couple of switches on the Cayenne console turn on and off the LEDs on the LoStik.

**Comment by WA4OSH** [comment]
It's very easy to setup a LoStik app on TTN and set its integration to Cayenne. It's also not hard to setup Cayenne.

**Comment by WA4OSH** [comment]
The other example would build an IFTTT example.

File: README.md [hunk 1]
```diff
@@ -32,6 +32,10 @@ will alternate the blue and red leds every .5 seconds.
 
 radio_receiver.py and radio_sender.py are used for sending LoRa packets between two LoRa Stiks without the need for a LoRaWAN gateway.  The included example sends a unix timestamp packet every 2 seconds and the receiver prints the incoming packets to stdout.
 
+### The Things Network
+
+lorawan_sender.py is used for sending LoRaWAN packets to a LoRaWAN gatewaywhich is connected to The Things Network backend. You need to create an application and pass the Application Key and the Application EUI as well as a Device EUI to the program.
```
Comment (lolsborn) [inline README.md]: missing a space on this line

**Comment by mbgg** [comment]
> ./lorawan_sender.py --appkey 7929233E8412329F0D9B2B20FF92043F --appeui [APP_EUI] --deveui [DEVEUI] --payload A1F3

you have to pass hexadecimal values using 0x prefix

File: examples/lorawan_sender.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    send_cmd("sys get hweui")
+    send_cmd("sys set pindig GPIO11 1")
+    send_cmd('sys get ver')
+    send_cmd('radio get mod')
+    send_cmd('radio get freq')
+    send_cmd('radio get sf')
+    send_cmd('mac pause')
+    send_cmd('mac resume')
+    send_cmd('mac reset 868')
```
[the remark this answers is not in the document]
Comment (mbgg) [reply examples/lorawan_sender.py]: I suppose the best would be to add a new argument, with one frequency as default. Which one do you prefer?

**Comment by lolsborn** [comment]
I just found the code that I had written to do LoRaWAN and created another PR here https://github.com/lolsborn/LoStik/pull/6 to compare.  I haven't had time to compare against your work, but my code does handle both otaa and abp and I've admittedly tested mostly abp.  When I get some time this week I'll compare them a bit closer and hopefully combine the best of them.

File: examples/lorawan_sender.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lolsborn) [reply examples/lorawan_sender.py]: Actually I'm not sure why pause, resume, and reset are necessary at all in this example?

**Comment by lolsborn** [comment]
I'm not merging this PR, but it was used along with the prior LoRaWAN branch I had started to provide the current example script which I know people have used successfully and I have been bug-fixing.  I appreciate the work here and did use bits of the code as reference to see how other people had implemented things.

**Pull request closed without merging**